### PR TITLE
Scikit learn fixes

### LINKED ISSUE #350: Adaboost and  Random Forest don't work with latest sklearn (1.2.0)
Latest sklearn deprecated some inputs and added input checks which are currently breaking Adaboost and RandomForest. The reason this hasn't been detected earlier is that many sklearn models are not covered by unit tests. 

I'm already working on a PR which will fix the issues and add test coverage for the missing sklearn models.

### PR DESCRIPTION
**Pull request by egillax**
fixes #350 

Also changed a few things with sklearn models: 
- Removed 'balanced' option from default set of hyperparameters from various models. Various models used these to put more weight on the minority class when optimizing. In light of @cynthiayang525 work on class imbalance I think its reasonable to not include those in the defaults. This reduces the search space by a factor of 2x. For example random forest goes from about 200 runs to 100 for the default set. 
- Changed ```epsilon``` in the sklearn MLP to one value in the defaults. This is a parameter for numerical stability during optimization and doesn't make sense to include it in hyperparameter tuning (or even changing it really). This reduces the search space by a factor of 3x for the MLP.

I updated the tests:
- create ```tinyTrainData``` in ```helper-objects``` to fit sklearn models and KNN quickly. This data only has the ten features with highest coefficients when fitting LASSO.
- Add fits of Adaboost, RandomForest, SVM, MLP and Naive-Bayes to tests.
- Added ```expect_correct_fitPlp``` expectation to test output of fitPLP for different classifiers. This function is in ```helper-expectations```. Since I was running the same tests for all the sklearn classifiers I refactored it out for ease of maintenance.
- Removed instances of ```createTrainData``` from tests. Now ```trainData``` is created once in the helpers and when we need new trainData to test on it's copied (using a new function ```copyTrainData```).

I did test this locally on sklearn versions 1.2.0, 1.1.2 and 1.0.2 to make sure my changes were backwards compatible.

### CODE REVIEW

**Comment by egillax** [comment]
Fixed seed issue for GBM as well (#347). Previous behavior was that no matter what seed was used it wasn't passed on to xgboost. It actually seems though the seed was fixed in xgboost internally because sequential runs always gave exactly same test auc. Now runs with same seed will give exactly same test AUC but different seeds will give slightly different results.

**Comment by jreps** [comment]
Thank you for finding these issues and fixing them :)

**Comment by egillax** [comment]
@jreps This is not completely ready to merge yet. I was trying to add a bit of tests to increase coverage and ensure the tests are stable. Want to test a bit more before I merge to ensure I'm not introducing tests that fail every 20 time they are run.

**Comment by egillax** [comment]
@jreps this should now be ready to merge. Added a few tests so the coverage increases by about 1%.

**Comment by jreps** [comment]
Thanks - this is great! :D
